File: packages/elixir-sdk/test/adapters/snapshots_test.exs
defmodule TreeDxSdk.SnapshotsAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Snapshots.build(client, "repo/a", %{})
    TreeDxSdk.Snapshots.get(client, "repo/a", "snap/a")

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :get and &1.path == "/api/v1/repos/repo%2Fa/snapshots/snap%2Fa")
           )
  end
end

File: packages/ts-sdk/test/adapters/storage/snapshots.test.ts
import { describe, expect, it } from 'vitest';
import { SnapshotsAdapter } from '../../../src/treedx/index.js';
import { MockTransport } from '../mock.js';

describe('SnapshotsAdapter', () => {
  it('constructs snapshot endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new SnapshotsAdapter({ transport });
    await adapter.build('repo', {});
    await adapter.get('repo', 'snap');
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'POST /api/v1/repos/repo/snapshots/build',
      'GET /api/v1/repos/repo/snapshots/snap'
    ]);
  });
});
